### ISSUE TITLE
pin used multiple times

### ISSUE THREAD

**Issue by beireken** [Erriez/ESPHomePCPowerControlHomeAssistant#12]
Recent esphome throws an error because the switch pin is used multiple times in the yaml.

I had to change the config to:

```
switch:
  - platform: gpio
    name: "PC Power Toggle"
    icon: "mdi:electric-switch"
    pin: 
      number: D1   # Power button output pin
      allow_other_uses: true
    id: power_short_press
    inverted: no
    on_turn_on:
    - delay: 150ms
    - switch.turn_off: power_short_press
  - platform: gpio
    name: "PC HARD POWER OFF"
    icon: "mdi:electric-switch"
    pin: 
      number: D1   # Power button output pin
      allow_other_uses: true
    id: power_long_press
    inverted: no
    on_turn_on:
    - delay: 3500ms
    - switch.turn_off: power_long_press
```

The it compiles fine.

**Comment by Erriez** [comment]
@beireken Thanks for using my project and time to report this issue. I was able to reproduce the problem and was not aware that the ESPHome `yaml` file format changed. Some more updates are required according to the documentation:
* `esp8266` or `esp32` moved to a separate line. (Was single line: `platform: ESP8266 # ESP8266 or ESP32`).
* Line `- platform: esphome` must be added to `ota` section.
* `inverted` option moved to `pin`.
* New pin `mode` configuration under `pin`.

I've created this version:

```yaml
esphome:
  name: pc-power

# ESP8266 board
esp8266:
  board: nodemcuv2

# Or: ESP32 board (comment out esp8266)
# esp32:
#   lolin_d32

# WiFi connection to base station
wifi:
  ssid: !secret esphome_wifi_ssid
  password: !secret esphome_wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "PC Power Fallback Hotspot"
    password: !secret esphome_ap_password

# captive_portal:

# Enable logging
logger:

# Enable Home Assistant API
api:
  password: !secret esphome_api_password

# Enable OTA update via WiFi
ota:
  - platform: esphome
    password: !secret esphome_ota_password

switch:
  - platform: gpio
    id: power_short_press
    name: "PC Power Toggle"
    icon: "mdi:electric-switch"
    pin:
      number: D2   # Power button output pin
      inverted: false
      allow_other_uses: true
      mode:
        output: true
    on_turn_on:
    - delay: 150ms
    - switch.turn_off: power_short_press
  - platform: gpio
    id: power_long_press
    name: "PC HARD POWER OFF"
    icon: "mdi:electric-switch"
    pin:
      number: D2   # Power button output pin
      inverted: false
      allow_other_uses: true
      mode:
        output: true
    on_turn_on:
    - delay: 3500ms
    - switch.turn_off: power_long_press

binary_sensor:
  - platform: gpio
    name: "PC Power State"
    device_class: power
    pin:
      number: D1   # Power detect input pin (readback from Reset button)
      inverted: false
      mode:
        input: true
```

```
$ esphome compile pc-power.yaml 
INFO ESPHome 2024.9.2
INFO Reading configuration pc-power.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing pc-power (board: nodemcuv2; framework: arduino; platform: platformio/espressif8266@4.2.1)
------------------------------------------------------------------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
Dependency Graph
|-- ESP8266WiFi @ 1.0
|-- ESP8266mDNS @ 1.2
RAM:   [====      ]  38.1% (used 31220 bytes from 81920 bytes)
Flash: [====      ]  35.2% (used 367639 bytes from 1044464 bytes)
====================================================== [SUCCESS] Took 0.90 seconds ======================================================
INFO Successfully compiled program.
```

Can you check if this version works? Then I'll update the yaml file.

Note: In your version you changed to pin D1 which should be D2 to match with the schematic.

Thanks!

**Comment by beireken** [comment]
Yes, your yaml works fine!

Not sure why i'm using the D1 pin  actually...i'm using this module for nearly a year i think.
I don't use the sensor part though, i forgot why, guess i got lazy at some point and just left it in the pc :)
I use a ping as my on sensor at the moment, :D

**Comment by Erriez** [comment]
Thanks for letting me know! I've merged the updates and released v1.0.1.
